internal/executor/podchecks/pod_checks.go

```diff
@@ -31,11 +33,18 @@ func NewPodChecks(cfg config.Checks) (*PodChecks, error) {
 		return nil, err
 	}
 
-	return &PodChecks{eventChecks: ec, containerStateChecks: csc}, nil
+	return &PodChecks{eventChecks: ec, containerStateChecks: csc, timeWithoutEventsOrStatus: cfg.TimeWithoutEventsOrStatus}, nil
 }
 
 func (pc *PodChecks) GetAction(pod *v1.Pod, podEvents []*v1.Event, timeInState time.Duration) (Action, string) {
 	messages := []string{}
+
+	isNodeBad := pc.isNodeBad(pod, podEvents)
```

Comment:
I'm in two minds on if this should just append the messages to the array instead of returning but functionally looks good

Reply:
If this is true than we really only have 1 entry in the array.  If we have no pod events or pod status, we shouldn't run the getAction of these other steps.  At least in my testing, we hit a panic.  

So I made sure that if the node is bad (and stuck) than we Retry and I set a message.